### ISSUE TITLE
Support optional parameters

### ISSUE THREAD

**Issue by bitwiseman** [josdejong/typed-function#6]
Once you support `undefined` having an easy shorthand for `undefined or null` as `?` would be good

For example: `string?`.

**Comment by josdejong** [comment]
Yes indeed, but besides that `typed-function` should be able to accept a function being called with a parameter less. Right now it would just throw an error "Too few arguments" when you leave out the last (optional) argument.
